mesh/ncmesh.cpp

```diff
@@ -3402,77 +3408,81 @@ void NCMesh::NCList::Clear()
       point_matrices[i].DeleteAll();
    }
 
-   inv_index.DeleteAll();
+   inv_index.clear();
+}
+
+NCMesh::NCList::MeshIdAndType
+NCMesh::NCList::GetMeshIdAndType(int index) const
+{
+   BuildIndex();
+   if (inv_index.size() == 0)
```

Comment:
Doesn't `BuildIndex` make this check unnecessary?

Reply:
This check is mainly so that these methods aren't called before the sublists have been populated. I moved the assert into the `BuildIndex` call in https://github.com/mfem/mfem/pull/3926/commits/c27bec417c62cb6dd7167dfa3939bbc1142fe31a.